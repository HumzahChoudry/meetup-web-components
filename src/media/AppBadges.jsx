import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';

export const IOS_DOWNLOAD_LINK = 'http://meetu.ps/2ZhShs';
export const ANDROID_DOWNLOAD_LINK = 'http://meetu.ps/2ZhSyb';
const getAppStoreDownloadPhoto = (platform, language) =>
	// eslint-disable-next-line global-require, import/no-dynamic-require
	require(`../../assets/images/app_download/${platform}/download_${language}.png`);

/**
 * Renders iOS and Android app badges
 * @class AppBadges
 */
const AppBadges = ({ className, language, getAppStorePhoto, ...other }) => {
	const classNames = cx('getTheApp-downloadLinks', className);

	return (
		<Flex className={classNames} {...other}>
			<FlexItem className="getTheApp-ios">
				<a href={IOS_DOWNLOAD_LINK}>
					<img
						className="getTheApp-downloadImage span--100 margin--center"
						src={getAppStorePhoto('ios', language)}
					/>
				</a>
			</FlexItem>
			<FlexItem className="getTheApp-android">
				<a href={ANDROID_DOWNLOAD_LINK}>
					<img
						className="getTheApp-downloadImage span--100 margin--center"
						src={getAppStorePhoto('android', language)}
					/>
				</a>
			</FlexItem>
		</Flex>
	);
};

AppBadges.defaultProps = {
	getAppStorePhoto: getAppStoreDownloadPhoto,
};

AppBadges.propTypes = {
	getAppStorePhoto: PropTypes.func,
	locale: PropTypes.string.isRequired,
};

export default AppBadges;
