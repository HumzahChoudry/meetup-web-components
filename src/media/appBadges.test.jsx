import React from 'react';
import { shallow } from 'enzyme';

import AppBadges, { IOS_DOWNLOAD_LINK, ANDROID_DOWNLOAD_LINK } from './AppBadges';

const testLocale = 'fr';

describe('App Badges', () => {
	const appBadgesComponent = shallow(<AppBadges locale={testLocale} />);
	it('exists', () => {
		expect(appBadgesComponent).toMatchSnapshot();
	});
	it('should link to the itunes store', () => {
		const itunesStoreLink = appBadgesComponent.find(`a[href="${IOS_DOWNLOAD_LINK}"]`);
		expect(itunesStoreLink.exists()).toBe(true);
	});
	it('should link to the google play store', () => {
		const googlePlayStoreLink = appBadgesComponent.find(
			`a[href="${ANDROID_DOWNLOAD_LINK}"]`
		);
		expect(googlePlayStoreLink.exists()).toBe(true);
	});
	it('should render the correct itunes and google play store images based on locale', () => {
		const getAppStorePhoto = jest.fn();
		shallow(<AppBadges locale={testLocale} getAppStorePhoto={getAppStorePhoto} />);
		expect(getAppStorePhoto).toHaveBeenCalledWith('android', testLocale);
		expect(getAppStorePhoto).toHaveBeenCalledWith('ios', testLocale);
	});
});
