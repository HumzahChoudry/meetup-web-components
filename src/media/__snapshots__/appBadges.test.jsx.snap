// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Badges exists 1`] = `
<Flex
  className="getTheApp-downloadLinks"
  direction="row"
>
  <FlexItem
    className="getTheApp-ios"
  >
    <a
      href="http://meetu.ps/2ZhShs"
    >
      <img
        className="getTheApp-downloadImage span--100 margin--center"
        src=""
      />
    </a>
  </FlexItem>
  <FlexItem
    className="getTheApp-android"
  >
    <a
      href="http://meetu.ps/2ZhSyb"
    >
      <img
        className="getTheApp-downloadImage span--100 margin--center"
        src=""
      />
    </a>
  </FlexItem>
</Flex>
`;
